File: backend/app/schemas/mcp_write_requests.py
from pydantic import BaseModel


class McpWriteRequestOut(BaseModel):
    id: str
    ontology_id: str
    release_id: str
    descriptor_id: str
    target_instance_id: str | None = None
    parameters: dict
    preview_hash: str
    preview_canonical: str
    status: str
    created_at: object
    resolved_at: object | None = None

File: frontend/src/api/mcpWriteRequests.ts
import { apiClient } from './client'

export interface McpWriteRequestItem {
  id: string
  ontology_id: string
  release_id: string
  descriptor_id: string
  target_instance_id: string | null
  parameters: Record<string, unknown>
  preview_hash: string
  preview_canonical: string
  status: string
  created_at: string
  resolved_at: string | null
}

export interface McpWriteRequestResolution {
  id: string
  status: string
}

export const mcpWriteRequestsApi = {
  list: () => apiClient.get<{ items: McpWriteRequestItem[] }>('/mcp/write-requests'),
  approve: (id: string) => apiClient.post<McpWriteRequestResolution>(`/mcp/write-requests/${id}/approve`),
  reject: (id: string) => apiClient.post<McpWriteRequestResolution>(`/mcp/write-requests/${id}/reject`),
}
